File: poker-server/src/main/java/com/poker/poker/models/game/DrawGameDataModel.java
package com.poker.poker.models.game;

import io.swagger.v3.oas.annotations.media.ArraySchema;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Schema(description = "Model representing player data in a game, i.e. cards drawn, etc...")
public class DrawGameDataModel {
  @Schema(implementation = GamePlayerModel.class)
  private GamePlayerModel player;

  @Schema(description = "Flag that is true if this draw is next.", example = "true")
  private boolean acting;

  @ArraySchema(schema = @Schema(implementation = DrawGameDrawModel.class))
  private List<DrawGameDrawModel> draws;
}

File: poker-client/src/app/api/models/draw-game-data-model.ts
/* tslint:disable */
import { DrawGameDrawModel } from './draw-game-draw-model';
import { GamePlayerModel } from './game-player-model';

/**
 * Model representing player data in a game, i.e. cards drawn, etc...
 */
export interface DrawGameDataModel {

  /**
   * Flag that is true if this draw is next.
   */
  acting?: boolean;
  draws?: Array<DrawGameDrawModel>;
  player?: GamePlayerModel;
}
